### ISSUE TITLE
SQL Server prepared statement setObject instrumentation error

### ISSUE THREAD

**Issue by sfthomas** [Microsoft/ApplicationInsights-Java#286]
```
java.lang.VerifyError: StackMapTable format error: bad type array size in method com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.setObject(ILjava/lang/Object;)V
    at com.microsoft.sqlserver.jdbc.SQLServerConnection.prepareStatement(SQLServerConnection.java:2167)
    at com.microsoft.sqlserver.jdbc.SQLServerConnection.prepareStatement(SQLServerConnection.java:1853)
    at org.apache.tomcat.dbcp.dbcp2.DelegatingConnection.prepareStatement(DelegatingConnection.java:287)
    at org.apache.tomcat.dbcp.dbcp2.DelegatingConnection.prepareStatement(DelegatingConnection.java:287)
    at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$5.doPrepare(StatementPreparerImpl.java:146)
    at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:172)
    at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareQueryStatement(StatementPreparerImpl.java:148)
    at org.hibernate.loader.Loader.prepareQueryStatement(Loader.java:1927)
    at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:1896)
    at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:1874)
    at org.hibernate.loader.Loader.doQuery(Loader.java:919)
    at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:336)
    at org.hibernate.loader.Loader.doList(Loader.java:2610)
    at org.hibernate.loader.Loader.doList(Loader.java:2593)
    at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2422)
    at org.hibernate.loader.Loader.list(Loader.java:2417)
    at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:501)
    at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:371)
    at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:216)
    at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1339)
    at org.hibernate.internal.QueryImpl.list(QueryImpl.java:87)
    at org.hibernate.jpa.internal.QueryImpl.list(QueryImpl.java:606)
    at org.hibernate.jpa.internal.QueryImpl.getResultList(QueryImpl.java:483)

```

Using Java 8, HIbernate, Tomcat 8, and the same error is coming for the MS JDBC4 Driver 6.0, 4.2 as well as for the JDBC42 drivers version 6.0 and 4.2.

**Comment by Dmitry-Matveev** [comment]
I would recommend to update to the latest version 1.0.7, and if the issue persists, disable JDBC instrumentation in applicationinsights.xml config.

**Comment by rh78** [comment]
I have an instrumentation error as well:

Caused by: java.lang.VerifyError: Expecting a stackmap frame at branch target 23
7
Exception Details:
  Location:
    com/microsoft/sqlserver/jdbc/SQLServerStatement.<init>(Lcom/microsoft/sqlser
ver/jdbc/SQLServerConnection;II)V @170: if_icmpeq
  Reason:
    Expected stackmap frame at this location.
  Bytecode:
    0x0000000: 2ab7 00cc 2a03 b500 682a 03b5 006a 2a03
    0x0000010: b500 b12a 01b5 009e 2a01 b500 7a2a 03b5
    0x0000020: 00b5 2a03 b500 a82a 03b5 00b3 2a14 00cd
    0x0000030: b500 d02a 03b5 00d2 2a03 b500 d42a 03b5
    0x0000040: 00d6 2abb 00d8 59b7 00d9 b500 dbb8 00dd

**Comment by dhaval24** [comment]
@rh78 we have noted this issue with the Agent in the latest JDK. Can you please have the look at #419 
I have pointed a work around to pass XVerify:none in tomcat startup script to avoid errors. We are investigating for permanent workaround.

**Comment by dhaval24** [comment]
fixed in #449
